pkg/userdata/userdata.go

```diff
@@ -215,11 +218,14 @@ coreos:
       Environment="RKT_OPTS=\
         --uuid-file-save=/var/run/kubelet-pod.uuid \
         --volume etc-resolv,kind=host,source=/etc/resolv.conf --mount volume=etc-resolv,target=/etc/resolv.conf \
+        --volume etc-cni,kind=host,source=/etc/cni --mount volume=etc-cni,target=/etc/cni \
+        --volume opt-cni,kind=host,source=/opt/cni,readOnly=true --mount volume=opt-cni,target=/opt/weave-net \
```

Comment:
This configuration is required only for weave as the weave binaries are not supplied with the default coreos hyperkube image and need to be mounted in specifically. The other binaries are supplied with the image and match with a working kubelet. See https://github.com/weaveworks/weave/issues/2613